From 5541a7e7712ddc5f2eaf4321675201f281610c85 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Thu, 17 Jul 2025 13:18:05 +0100
Subject: [PATCH 04/23] binutils: drop unused note_size, contents, old
 variables

GCC trunk recently had improvements to its -Wunused-but-set-variable which
picked up that contents and hence note_size & old aren't used at all in the
end.

	* objcopy.c (merge_gnu_build_notes): Drop unused 'note_size', 'contents',
	and 'old' variables.

(cherry picked from commit d3d1718602b179d8ec833851f5cb67ab9d314a92)
---
 binutils/objcopy.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/binutils/objcopy.c b/binutils/objcopy.c
index 038f6555c0b..905ce917708 100644
--- a/binutils/objcopy.c
+++ b/binutils/objcopy.c
@@ -2529,7 +2529,6 @@ merge_gnu_build_notes (bfd *          abfd,
 
   /* Reconstruct the ELF notes.  */
   bfd_byte *     new_contents;
-  bfd_byte *     old;
   bfd_byte *     new;
   bfd_vma        prev_start = 0;
   bfd_vma        prev_end = 0;
@@ -2537,12 +2536,8 @@ merge_gnu_build_notes (bfd *          abfd,
   /* Not sure how, but the notes might grow in size.
      (eg see PR 1774507).  Allow for this here.  */
   new = new_contents = xmalloc (size * 2);
-  for (pnote = pnotes, old = contents;
-       pnote < pnotes_end;
-       pnote ++)
+  for (pnote = pnotes; pnote < pnotes_end; pnote ++)
     {
-      bfd_size_type note_size = 12 + pnote->padded_namesz + pnote->note.descsz;
-
       if (! is_deleted_note (pnote))
 	{
 	  /* Create the note, potentially using the
@@ -2585,8 +2580,6 @@ merge_gnu_build_notes (bfd *          abfd,
 	      prev_end = pnote->end;
 	    }
 	}
-
-      old += note_size;
     }
 
 #if DEBUG_MERGE
-- 
2.49.1

